### ISSUE TITLE
bug "This key pair is invalid! Please select keys for different parties."

### ISSUE THREAD

**Issue by KuzMi13** [sh-dv/hat.sh#97]
When entering the Public and Private key, it does not allow you to take the next step and writes "This key pair is invalid! Please select keys for different parties."

**Comment by x71coment** [comment]
That's right, after generating keys on the site and saving them to disk, during encryption after selecting these keys, it turns out that these keys do not match.

**Comment by KuzMi13** [comment]
How to make it work correctly?

Вс, 14 янв. 2024 г. в 20:58, x71coment ***@***.***>:

**Comment by mrtechtroid** [comment]
Hi @KuzMi13 and @x71coment, 
When using asymmetric encryption, you are supposed to use your private key, and your recipients public key, and not your own public key. Have a watch at this [video](https://www.youtube.com/watch?v=AQDCe585Lnc) to know more.

**Comment by fakelauzenhao** [comment]
![Image](https://github.com/user-attachments/assets/0a9f46c4-df89-4065-9c4f-68ea4b1d2c92)
